cookbooks/bcpc-hadoop/recipes/disks.rb

```diff
@@ -47,17 +47,28 @@
       # What disks will bcpc and bcpc-hadoop feel free to blank?
       # By default, all sd* devices (excluding sda) and all md* devices.
       #
-      # On our EFI-based VM builds, it's very important to omit sdb, as
-      # that is the 32 MB image containing iPXE.  (It's relatively
-      # harmless to overwrite it, but it will cause graphite to fail when
-      # /disk/0 fills up.)
+      # On our EFI-based VM builds, it's very important to the 32 MB
+      # image containing iPXE.  (It's relatively harmless to overwrite
+      # it, but it will cause graphite to fail when /disk/0 fills up.)
+      #
+      # We also reject any block device we are unable to open with O_EXCL,
+      # because that means it is already in use by the kernel.
       #
       all_drives = node[:block_device].keys.select do |dd|
-        dd =~ /sd[a-i]?[b-z]/ ||
-        dd =~ /md\d+/
+        dd =~ /sd[a-i]?[a-z]/ || dd =~ /md\d+/
```

Comment:
Ah my mental regex parsing was faulty; indeed `[a-i]?` is zero or one `a-i`'s while the trailing `[a-z]` will cover `/dev/sd[a-z]` which I was failing to grok.